### DISCUSSION TITLE
409 problem

### DISCUSSION THREAD

**Discussion by Stiropor** [inertiajs/inertia#498] [Help (Vue)]
Sorry for asking here but I cannot seem to find an answer anywhere :(

Intertia works nice, it's amazing, I've just build pretty big (internal) app with it, I love this library.
But after updating to latest version 2 weeks ago I'm having a problem with versions ... when developing in Laravel and I change something in vue file it gets rebuild and all. When I click on a link (and not reload) I get `TypeError: can't access property "x-inertia", t.headers is undefined`. I do get  "X-Inertia-Location" header but no x-inertia header anywhere. I have latest versions (using Vue2 still), added middleware in Laravel and all that.

How do I make it refresh when assets change (meaning getting 409 response) and how do I get this X-inertia header to be sent, or am I missing something? At least from website and github posts I understood that Inertia should reload page when 409 header is returned? If not, how can I make it work this waY

This is how I create vue instance
```
new Vue({
	render: h => h(InertiaApp, {
		props: {
			initialPage: JSON.parse(app.dataset.page),
			resolveComponent: name => require(`./Pages/${name}`).default,
		},
	}),
}).$mount(app)
```
which is basic, Laravel's middleware is unchanged. What am I missing?

**Comment by oidre** [comment]
just add versioning in your generated app.js file.
in webpack.mix.js add .version()

ref: [https://laravel.com/docs/8.x/mix#versioning-and-cache-busting](https://laravel.com/docs/8.x/mix#versioning-and-cache-busting)

**Reply by Stiropor** [reply]
I have that by default already. It probably has something to do with x-inertia head not being sent :/

**Comment by kevindees** [comment]
If you are coming from Inertia `v1` and moving to `v2`, you will get this error if your package dependencies are using an older version and the `Link` from `import {Link} from '@inertiajs/vue3';` will not send the `x-inertia-version`. Be sure all dependencies are on the same version of inertia.
